test-app/tests/integration/components/multiselect-test.gts

```diff
@@ -1501,6 +1502,134 @@ module('Integration | Component | Multiselect', function (hooks) {
     assert.dom('[role="option"]').exists({ count: 2 });
   });
 
+  test('it renders the `Select all` option when provided with `@selectAllText`', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @noResultsText="No results"
+        @selectAllText="Select all"
+        data-multiselect
+      >
+        <:chip as |chip|>
+          <chip.Chip>
+            {{chip.option}}
+            <chip.Remove @label="Remove" />
+          </chip.Chip>
+        </:chip>
+
+        <:default as |multiselect|>
+          <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+        </:default>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[data-multiselect-select-all-option]').exists();
+    assert.dom('[data-multiselect-select-all-option]').hasText('Select all');
+
+    assert.dom('[data-multiselect-select-all-checkbox]').exists();
+  });
+
+  test('it handles all possible state changes when the `Select all` option is interacted with', async function (assert) {
+    assert.expect(22);
+
+    class State {
+      @tracked selected: string[] = [];
+    }
+
+    let state = new State();
```

Comment:
We have to use `State` here because we need `@selected` passed into `Multiselect` to be `@tracked`; otherwise, our value will not update when the `handleChange` sets the new value. So a bit more complicated than our other tests.